spec/index.bs

```diff
@@ -2517,6 +2518,82 @@ dictionary GPUImageBitmapCopyView {
+  - |destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/y}} must be a multiple of [=texel block height=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |copySize|.{{GPUExtent3DDict/width}} must be a multiple of [=texel block width=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |copySize|.{{GPUExtent3DDict/height}} must be a multiple of [=texel block height=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |source|.{{GPUBufferCopyView/bytesPerRow}} must be a multiple of 256.
+  - If |source|.{{GPUBufferCopyView/bytesPerRow}} is not 0, it must be greater than or equal to the number of |bytes in a complete row|.
+  - If |source|.{{GPUBufferCopyView/bytesPerRow}} is 0, the number of |bytes in a complete row| must be a multiple of 256.
+  - |destination|.{{GPUTextureCopyView/mipLevel}} must be less than the {{GPUTexture/[[mipLevelCount]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |destination|.{{GPUTextureCopyView/arrayLayer}} must be less than the {{GPUTexture/[[textureSize]]}}.{{GPUExtent3DDict/depth}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - {{GPUExtent3DDict/depth}} must be 1 when the {{GPUTexture/[[dimension]]}} of {{GPUTextureCopyView/texture}} is {{GPUTextureDimension/2d}}.
+  - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/x}} + |copySize|.{{GPUExtent3DDict/width}}) must not overflow a {{GPUIntegerCoordinate}}.
+  - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/y}} + |copySize|.{{GPUExtent3DDict/height}}) must not overflow a {{GPUIntegerCoordinate}}.
+  - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/z}} + |copySize|.{{GPUExtent3DDict/depth}}) must not overflow a {{GPUIntegerCoordinate}}.
+  - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/x}} + |copySize|.{{GPUExtent3DDict/width}}) must be less than or equal to the {{GPUExtent3DDict/width}} of the [=physical size=] of
+    |destination|.{{GPUTextureCopyView/texture}} [=subresource=] at [=mipmap level=] {{GPUTextureCopyView/mipLevel}} and the [=array layer=] {{GPUTexture/[[textureSize]]}}.{{GPUExtent3DDict/depth}}
+    when the {{GPUTexture/[[dimension]]}} of {{GPUTextureCopyView/texture}} is {{GPUTextureDimension/2d}}.
+  - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/y}} + |copySize|.{{GPUExtent3DDict/height}}) must be less than or equal to the {{GPUExtent3DDict/width}} of the [=physical size=] of
+    |destination|.{{GPUTextureCopyView/texture}} [=subresource=] at [=mipmap level=] {{GPUTextureCopyView/mipLevel}} and the [=array layer=] {{GPUTexture/[[textureSize]]}}.{{GPUExtent3DDict/depth}}
+    when the {{GPUTexture/[[dimension]]}} of {{GPUTextureCopyView/texture}} is {{GPUTextureDimension/2d}}.
+  - |source|.{{GPUBufferCopyView/buffer}}.{{GPUBuffer/[[size]]}} must be greater than or equal to |source|.{{GPUBufferCopyView/offset}}.
```

Comment:
I think this could be merged with the rule below saying that size must be greater than or equal to the offset + required bytes to copy

Reply:
Agreed, and flip it so it says "source.offset + required bytes in copy must be <= source.buffer.[[size]]".

Don't write this to guard against overflow; it's math, not code.

Reply:
Done